### ISSUE TITLE
how to increase memory for GAP 4.13.0 under windows

### ISSUE THREAD

**Issue by qodesign** [gap-system/gap#5687]
I've been running GAP under windows for years. 
In the older versions, increasing memory was straight forward using "gap.exe -m 16000m"
but the latest GAP uses mintty and several layers of .sh files. (run-gap-mintty.sh,bash /run-gap.sh,...)
I'm not sure where the "-m 1600m" equivalent would be inserted.

```
 ┌───────┐   GAP 4.13.0 of 2024-03-15
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-cygwin-default64-kv9
 Configuration:  gmp 6.3.0, GASMAN, readline
 Loading the library and packages ...
 Packages:   AClib 1.3.2, Alnuth 3.2.1, AtlasRep 2.1.8, AutPGrp 1.11, Browse 1.8.21, CaratInterface 2.3.6,
             CRISP 1.4.6, Cryst 4.1.27, CrystCat 1.1.10, CTblLib 1.3.9, curlInterface 2.3.2, FactInt 1.6.3, FGA 1.5.0,
             Forms 1.2.9, GAPDoc 1.6.7, genss 1.6.8, IO 4.8.2, IRREDSOL 1.4.4, LAGUNA 3.9.6, orb 4.9.0,
             Polenta 1.3.10, Polycyclic 2.16, PrimGrp 3.4.4, RadiRoot 2.9, recog 1.4.2, ResClasses 4.7.3,
             SmallGrp 1.5.3, Sophus 1.27, SpinSym 1.5.2, StandardFF 1.0, TomLib 1.2.11, TransGrp 3.6.5, utils 0.85
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'

```

**Comment by ChrisJefferson** [comment]
The file you want to edit is the last line of `runtime/run-gap.sh`.

However, you might also want to check if you still need to adjust GAP's memory usage. 

In the standard GAP release, I made a list of size `16000*1024*1024/8` (the /8 is because each GAP object takes 8 bytes), this made GAP take about 15GB of memory, but I didn't get any warnings or issues.

**Comment by qodesign** [comment]
I edited the last line of runtime/run-gap.sh to /opt/gap-4.13.0/gap -m 16000m
gap am up but I don't know how to check that it actually accepted that...other than run the program that crashed before.
As an experiment I add a random string /opt/gap-4.13.0/gap -abcdefxxx 1 and gap also ran without complaining; this seems to suggest that the "-m 16000m" and the junk arguments are not even parsed. Either way GAP documentation should clearly say how this is to be done and if there's a way to verify the allocated memory.

**Comment by ChrisJefferson** [comment]
Are you sure you saved the file (if you installed globally in program files, you'll need admin access to write).

I did what you describe, and I get output:
```
gap: sorry, options must not be grouped '-abcxyzzz'.
usage: gap [OPTIONS] [FILES]
       run the Groups, Algorithms and Programming system, Version 4.13.0
       use '-h' option to get help.

GAP exited with an error. Press Enter to close window
```

If you want to see the current limits, you can run `GasmanLimits();`.

**Comment by qodesign** [comment]
I double checked and the file definitely gets saved with the edits.
I sill see the same behavior : -m 16000m has no affect and -abcxyzzz does not results in any complaints...GAP comes up the same way in both cases. GasmanLimits() gives the same value in all cases...strange that we're seeing different things. I'm running Windows 10 Pro; maybe a third Windows user can shed light on this

**Comment by ChrisJefferson** [comment]
Just checking what's different between our machines:

I run GAP by clicking on the start menu icon. When I right-click on it, select "open file location", the right-click on the GAP icon in the resulting directory and select "details", it a shortcut to target: C:\Program Files\GAP-4.13.0\gap-mintty.bat , which Starts in C:\Program Files\GAP-4.13.0

If I go into C:\Program Files\GAP-4.13.0 and run gap-mintty.bat , then it opens the gap in that directory, and edits I make to 'run-gap.sh' inside runtime in that directory change how GAP runs.

**Comment by qodesign** [comment]
I use a different approach but I think I found the source of the problem. Windows lets me edit run-gap.sh in its original directory even though I'm using non-admin account; I even see the changes in the file but somehow GAP behaves as if no changes are made. I found a solution by moving run-gap.sh out of that directory; Windows then asks for admin privilege. I make the edits where I moved it and then move it back to the original directory (Windows asks for admin privilege again). Then the edits now do have an affect and I can change the memory...(the junk option also makes GAP exit as expected) . So this seems like a solution that works around Window's oddities. Thanks for your help.
